modules/rosetta/src/main/scala/org/tessellation/rosetta/http/routes/ConstructionRoutes.scala

```diff
@@ -27,6 +28,15 @@ final class ConstructionRoutes[F[_]: Async](
           .bimap(_.toRosettaError, ConstructionDerive.Response(_))
           .asRosettaResponse
       }
+
+    case req @ POST -> Root / "hash" =>
+      req.decodeRosettaWithNetworkValidation[ConstructionHash.Request](appEnvironment, _.networkIdentifier) { hashReq =>
+        constructionService
+          .deserializeTransactionHashFromHex(hashReq.signedTransaction)
+          .bimap(_.toRosettaError, hash => ConstructionHash.Response(TransactionIdentifier(hash)))
```

Comment:
actually service could return a transaction identifier and the name of the method could be: `getTransactionIdentifier`,

so it would be:
```scala
constructionService
  .getTransactionIdentifier(hashReq.signedTransaction)
  .bimap(_.toRosettaError, ConstructionHash.Response(_))
  ...
```
```

Reply:
Changed.